public/app/core/components/connectWithCleanUp.tsx

```diff
@@ -0,0 +1,36 @@
+import { MapStateToPropsParam, MapDispatchToPropsParam, connect, useDispatch } from 'react-redux';
+import { StateSelector, cleanUpAction } from '../actions/cleanUp';
+import React, { ComponentType, FunctionComponent, useEffect } from 'react';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+export const connectWithCleanUp = <
+  TStateProps extends {} = {},
+  TDispatchProps = {},
+  TOwnProps = {},
+  State = {},
+  TSelector extends object = {}
+>(
+  mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps, State>,
+  mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>,
+  stateSelector: StateSelector<TSelector>
+) => (Component: ComponentType<any>) => {
+  const ConnectedComponent = connect(
+    mapStateToProps,
+    mapDispatchToProps
+  )(Component);
+
+  const ConnectedComponentWithCleanUp: FunctionComponent = props => {
+    const dispatch = useDispatch();
+    useEffect(() => {
+      return () => dispatch(cleanUpAction({ stateSelector }));
+    });
```

Comment:
Probably should have `[]` as a dependency so it is run only on unmount.

Reply:
Exactly, these tiny little bits about hooks...

Reply:
Good catch, will change